inspirehep/dojson/hep/receivers.py

```diff
@@ -26,15 +26,14 @@
 
 from itertools import chain
 
-from invenio_records.signals import before_record_insert, before_record_update
+from invenio_indexer.signals import before_record_index
 
 from inspirehep.utils.date import create_earliest_date
 from inspirehep.utils.helpers import force_force_list
 from inspirehep.utils.record import get_value
 
 
-@before_record_insert.connect
-@before_record_update.connect
+@before_record_index.connect
 def earliest_date(sender, *args, **kwargs):
```

Comment:
...And consequently stuff should be modified on `json`, not on `sender`, in the body of the method.